src/prefect/_internal/concurrency/api.py

```diff
@@ -90,6 +90,20 @@ def supervise_call_in_worker_thread(
         supervisor.submit()
         return supervisor
 
+    @staticmethod
+    def supervise_call_in_supervising_thread(
```

Comment:
All the changes in the `concurrency` module should be cherry-picked into another branch for review